src/main/java/com/aws/greengrass/mqttclient/AwsIotMqttClient.java

```diff
@@ -40,15 +46,27 @@
 class AwsIotMqttClient implements Closeable {
     static final String TOPIC_KEY = "topic";
     private static final String QOS_KEY = "qos";
+    private static final Random RANDOM = new Random();
     private final Logger logger = LogManager.getLogger(AwsIotMqttClient.class).createChild()
             .dfltKv(MqttClient.CLIENT_ID_KEY, (Supplier<String>) this::getClientId);
 
+    private final ExecutorService executorService;
     private final Provider<AwsIotMqttConnectionBuilder> builderProvider;
     @Getter
     private final String clientId;
     private MqttClientConnection connection;
     private final AtomicBoolean currentlyConnected = new AtomicBoolean();
     private final CallbackEventManager callbackEventManager;
+    private final Consumer<MqttMessage> messageHandler;
+    private final Topics mqttTopics;
+    @Getter(AccessLevel.PACKAGE)
+    private final Map<String, QualityOfService> subscriptionTopics = new ConcurrentHashMap<>();
+    private final Map<String, QualityOfService> droppedSubscriptionTopics = new ConcurrentHashMap<>();
+    private Future<?> resubscribeFuture;
+    @Setter
+    private static long waitTimeToSubscribeAgainMillis = Duration.ofMinutes(2).toMillis();
```

Comment:
NIT: make this a constant. Also why 2 minutes? Probably want to add a comment to explain that.

Reply:
Needed to set this to a tiny value for testing purposes so didn't make it `final`.

Open to discussion what's the appropriate delay. I'm using 2 minutes because I just copied `IotJobHelper`'s subscription retry delay.